runtime/frequency/src/lib.rs

```diff
@@ -146,17 +149,210 @@ pub type Executive = frame_executive::Executive<
+			if storage_result.is_some() {
+				let from: AccountId = storage_result.unwrap();
+				let account_data: AccountInfo<Index, AccountData<Balance>> = System::account(&from);
+				log::info!("🟢        Current Sudo account balance: free: {:?}, reserved: {:?}, fee_frozen: {:?}, misc_frozen: {:?}",
+					account_data.data.free, account_data.data.reserved, account_data.data.fee_frozen, account_data.data.misc_frozen);
+				from.clone()
+			} else {
+				log::warn!("This sudo key does not exist");
+				AccountId::from([0; 32])
+			}
+		}
+
+		// checks sudo balance post_upgrade, using the sudo AccountId passed from pre_upgrade results.
+		#[cfg(feature = "try-runtime")]
+		fn check_sudo_balance_post_upgrade(state: &Vec<u8>) {
+			// try_from is really the only way to try to get an AccountId[32] out of a byte vec;
+			// everything else is cfg-ed away in runtime.
+			match AccountId32::try_from(state.as_slice()) {
+				Ok(from) => {
+					let account_data: AccountInfo<Index, AccountData<Balance>> =
+						System::account(&from);
+					log::info!("‼️        Post-upgrade, Sudo account has a balance: free: {:?}, reserved: {:?}", account_data.data.free, account_data.data.reserved);
+				},
+				Err(_) => {
+					log::info!(
+						"❎        Post-upgrade, Sudo account could not be converted: {:?}",
+						state.as_slice()
+					);
+				},
+			}
+		}
+
+		// returns true if both Sudo storage keys are found, false otherwise.
+		fn keys_exist() -> bool {
+			let sudo_key: Vec<u8> = Self::join_keys(&SUDO_PREFIX, &KEY_PREFIX);
+			let sudo_pallet_version: Vec<u8> =
+				Self::join_keys(&SUDO_PREFIX, &PALLET_VERSION_PREFIX);
+
+			let mut keys_exist = frame_support::storage::unhashed::exists(sudo_key.as_slice());
+			log::info!("❓       Sudo Key storage exists ===>  {:?}", keys_exist);
+			keys_exist = frame_support::storage::unhashed::exists(sudo_pallet_version.as_slice());
+			log::info!("❓       Sudo PalletVersion storage exists ===>  {:?}", keys_exist);
+			keys_exist
+		}
+
+		#[cfg(not(feature = "frequency"))]
+		fn transfer_sudo_balance_to_treasury() {
+			log::warn!(
+				"‼️        transfer_sudo_balance_to_treasury was called but should not have been"
+			);
+		}
+
+		// TODO: correct weight
+		fn weights_from(reads: u64, writes: u64) -> Weight {
+			Weight::from_ref_time(0u64)
+				.saturating_add(RocksDbWeight::get().reads(reads))
+				.saturating_add(RocksDbWeight::get().writes(writes))
+		}
+
+		fn remove_storage(key: &Vec<u8>) {
+			frame_support::storage::unhashed::kill(key.as_slice());
+		}
+	}
+
+	impl OnRuntimeUpgrade for RemoveSudo {
+		// on_runtime_upgrade is the only OnRuntimeUpgrade trait function that must be defined for all configs
+		// do nothing if we are not on mainnet.
+		#[cfg(not(feature = "frequency"))]
+		fn on_runtime_upgrade() -> Weight {
+			Weight::zero()
+		}
+
+		// Do this if we are on mainnet
+		#[cfg(feature = "frequency")]
+		fn on_runtime_upgrade() -> Weight {
+			// TODO: move storage removal to fn in struct above
```

Comment:
seems done?

Reply:
Wasn't in my local.  It must not have been pushed.